### PR TITLE
feat: implement lambda parser

### PR DESCRIPTION
**Pull request by JosuePrado**
## Description

Provide a small description of Pull Request.

## Type of Change

Put an `x` in all the boxes that apply:

- [x] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [ ] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by fabianromeroclaros — COMMENTED** [review]
Add unit test to see the implementation in a better way please.

File: src/Parser/ParserExpresions.hs [hunk 1]
```diff
@@ -21,4 +27,8 @@ parseFunctionCall =
 parseBinaryExpr :: Parser Expr
 parseBinaryExpr = BinaryExpr <$> parseValue <*> (whiteSpaces *> parseArithmeticOperator <* whiteSpaces) <*> parseValue
 
+parseLambda :: Parser Expr
+parseLambda = LambdaExpr 
+              <$> (parseLambdaSymbol *> whiteSpaces *> parseOpenParents *> parseManyTypedIdentifier <* parseCloseParents <* whiteSpaces)
+              <*> parseLetStatement
```
Comment (teran-joseluis) [inline src/Parser/ParserExpresions.hs:33]: Why are you using `parseLetStatement` at the end?

**Review by SebasthianSalpozz — CHANGES_REQUESTED** [review]
Lamba is already reserved in reserved names, you should use that one and not create a new one to do it
Just use one of the two

**Review by teran-joseluis — CHANGES_REQUESTED** [review]

File: src/Parser/LexerParser.hs [hunk 2]
```diff
@@ -64,3 +64,5 @@ parseCloseParents = char ')'
 parseComma :: Parser Char
 parseComma = char ','
 
+parseLambdaSymbol :: Parser String 
+parseLambdaSymbol = string "lambda"
```
Comment (teran-joseluis) [inline src/Parser/LexerParser.hs:68]: We have lambda in the `languageDef`, you should review that.

**Review by teran-joseluis — CHANGES_REQUESTED** [review]

**Review by SebasthianSalpozz — APPROVED** [review]
Good Job!! :D

**Review by fabianromeroclaros — APPROVED** [review]

**Review by teran-joseluis — APPROVED** [review]
Good job!

**Pull request merged**